### ISSUE TITLE
[quotes] Backticks for union types

### ISSUE THREAD

**Issue by mordv** [typescript-eslint/typescript-eslint#3384]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**
```JSON
{
    "quotes": "off",
    "@typescript-eslint/quotes": [
      "warn",
      "backtick"
    ],
}
```

```TS
type A = `a` | 'b';
```

**Expected Result**
Either warning for `'b'` or separate rule for union types.

**Actual Result**
No warning for `'b'`

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `^4.22.0` |
| `@typescript-eslint/parser`        | `^4.22.0 `|
| `TypeScript`                       | `^4.1.2` |
| `ESLint`                           | `^7.25.0` |
| `node`                             | `^12.0.0` |

**Comment by bradzacher** [comment]
The reason that the rule works like this is because you *must* be on TS 4.1 for it to be safe to always use backticks for string literal types.
https://www.typescriptlang.org/play?ts=4.0.5#code/C4TwDgpgBAKgjFAvFA3lAhgLigZ2AJwEsA7AcygF8AoUSWJWOAbQAN0WBdAbiA

I think it's too soon for us to change this, or if we do change this it should be a breaking change.

**Comment by bradzacher** [comment]
Hi there 👋!

As you may have heard [ESLint core is deprecating its formatting rules](https://eslint.org/blog/2023/10/deprecating-formatting-rules/). We are likewise moving in the same direction and deprecating our formatting rules.

Deprecated rules will be removed in the next major version as their new home is the [eslint-stylistic](https://github.com/eslint-stylistic/eslint-stylistic) project.

If this issue is important to you please consider filling an issue over at eslint-stylistic which will be the new home for stylistic/formatting rules.

Thanks!
